File: backend/src/main/java/com/back/domain/care/dto/request/CareRequestDto.java
package com.back.domain.care.dto.request;


import com.back.domain.applicant.dto.request.ApplicantRequestDto;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDateTime;
import lombok.Builder;
import org.springframework.context.annotation.Description;

@Builder
public record CareRequestDto(
        @NotNull
        Long petId,

        @NotNull
        String title,

        ApplicantRequestDto applicantInfo,

        String anotherPets,

        String experience,

        String message,

        @NotNull
        LocalDateTime desiredStartDate,

        @Description("요청자가 원하는 돌봄 종료 날짜 - 무기한이 가능하므로 비어있을 수 있음")
        LocalDateTime desiredEndDate
) {

}

File: backend/src/main/java/com/back/domain/care/dto/request/CareRequestDto.kt
import com.back.domain.applicant.dto.request.ApplicantRequestDto
import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime

data class CareRequestDto(
    @field:NotNull
    val petId: Long = 0L,

    @field:NotNull
    val title: String = "",

    val applicantInfo: ApplicantRequestDto = ApplicantRequestDto(),

    val anotherPets: String? = null,

    val experience: String? = null,

    @field:NotNull
    val message: String = "",

    @field:NotNull
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val desiredStartDate: LocalDateTime = LocalDateTime.now(),

    /**
     * 요청자가 원하는 돌봄 종료 날짜 - 무기한이 가능하므로 비어있을 수 있음
     */
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val desiredEndDate: LocalDateTime? = null
)
